### DISCUSSION TITLE
Flashing fails when adding USE_REDUNDANT_FLASH_LAYOUT = "1"

### DISCUSSION THREAD

**Discussion by 0nlytom** [OE4T/meta-tegra#1556] [Q&A] [answered]
Hi all,

Not sure if I'm doing something wrong, but when adding Swupdate according to [this](https://github.com/OE4T/tegra-demo-distro/blob/master/layers/meta-tegrademo/dynamic-layers/meta-swupdate/README.md), the flashing stops at writing the primary partition (see below). Everything else builds ok  (SDK and Image). 

```
[ 210.7378 ] Writing partition APP with bla-bla-image.ext4.img [ 7094677940 bytes ]
Error: Return value 3                                         ] 000%
Command tegradevflash_v2 --pt flash.xml.bin --create
```
When I remove the lines enabling Swupdate, the device flashes without any issues. 

This is for `jetson-xavier-nx-devkit-emmc` and using the `scarthgap` branch of OE4T/mete-tegra. The carrier broad is a custom, however, I have not had any issues in the past year.

**Comment by dwalkes** [comment, marked as answer]
> Writing partition APP with bla-bla-image.ext4.img [ 7094677940 bytes ]

Did you confirm everything works as expected with `demo-image-base` and it's specific to `bla-bla-image`?

I'm guessing your image size is the issue.  Error handling for out of space issues is kind of obscure with the flash scripts currently and you'll be at the limit of a 16GB EMMC with this size and a dual rootfs setup I suspect.

**Reply by 0nlytom** [reply]
Did a total clean build, and it resolved the issue. It was for the `demo-image-base` as well as other images. Tested it on a Orin AGX with same results of failing to flash.

Thank you though
